open and write in byte mode using fileserver

gtmanfred opened the pull request:
### What does this PR do?
This switches to byte mode explicitely in the last few places I could find that
are using the fileserver payload to store files.

We should only ever be reading in the file, and then passing it to the minion,
the unicode locale on the master should not matter.  Then we want to make sure
to write the information in bytes to the filesystem.

gtmanfred committed:
open and write in byte mode using fileserver

This switches to byte mode explicitely in the last few places I could find that
are using the fileserver payload to store files.

We should only ever be reading in the file, and then passing it to the minion,
the unicode locale on the master should not matter.  Then we want to make sure
to write the information in bytes to the filesystem.

salt-jenkins commented:
@gtmanfred, thanks for your PR! By analyzing the history of the files in this pull request, we identified @terminalmage, @s0undt3ch and @shadowfax-chc to be potential reviewers.

s0undt3ch commented on salt/fileserver/hgfs.py:
```diff
@@ -314,7 +314,7 @@ def init():
-            with salt.utils.fopen(remote_map, 'w+') as fp_:
+            with salt.utils.fopen(remote_map, 'wb+') as fp_:
                 timestamp = datetime.now().strftime('%d %b %Y %H:%M:%S.%f')
                 fp_.write('# hgfs_remote map as of {0}\n'.format(timestamp))
```
Since you now opened in binary mode, you can only write bytes to the file. Since this is ASCII, it's safe to do `b'ascii string'`. On py3 of course.

gtmanfred replied:
Sorry, i missed reverting this one.

The only ones that needed to be changed was the env_cache which used the Serial output.

Thanks,
Daniel

terminalmage replied:
`salt.utils.to_bytes()` is a shorthand for this.

terminalmage replied:
Oh you're going to revert it

gtmanfred committed:
remove byte

This does not need to be bytes because we are only storing the hashes, and not
user files

rallytime commented:
I have switched the base branch on this pull request from `nitrogen` to `2017.7` in preparation for moving from a code-named release to a version number.

rallytime commented:
Go Go Jenkins!

rallytime commented:
Go Go Jenkins!

rallytime commented:
Go Go Jenkins!

rallytime commented:
Go Go Jenkins!

cachedout commented:
@s0undt3ch and @terminalmage Re-review requested.

cachedout commented:
Just waiting on @s0undt3ch for one more review and then we should be set.

rallytime commented:
Go Go Jenkins!
